test/image/mocks/ticklabelposition-1.json

```diff
@@ -0,0 +1,149 @@
+{
+    "data": [{
+        "xaxis": "x",
+        "yaxis": "y",
+        "x": ["20-12-15", "21-01-15"],
+        "y": [1e-1, 1e+6]
+    }, {
+        "xaxis": "x2",
+        "yaxis": "y2",
+        "x": ["20-10", "21-06"],
+        "y": [1e-1, 1e+6]
+    }, {
+        "xaxis": "x3",
+        "yaxis": "y3",
+        "x": ["20", "23"],
+        "y": [1e-1, 1e+6]
+    }, {
+        "xaxis": "x4",
+        "yaxis": "y4",
+        "x": ["01", "10"],
+        "y": [1e-1, 1e+6]
+    }],
+    "layout": {
+        "xaxis": {
+            "anchor": "y",
+            "domain": [0, 0.475],
+            "type": "date",
+            "ticklabelmode": "period",
+            "side": "bottom",
+            "ticks": "inside",
+            "ticklabelposition": "inside",
```

Comment:
These labels look a little off-center?
<img width="278" alt="Screen Shot 2020-11-22 at 9 17 48 AM" src="https://user-images.githubusercontent.com/2678795/99906182-a8e7ff80-2ca3-11eb-82aa-355bd219f86c.png">

Reply:
Please note that the mode of labels on this axis is *period*. So the labels are not centered on the ticks.
Resolving...